src/pages/Signup.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable react/jsx-props-no-spreading */
+import { useForm } from 'react-hook-form';
+import { useDispatch } from 'react-redux';
+import { useNavigate, Link } from 'react-router-dom';
+import { signupUser } from '../redux/actions/auth';
+
+const Signup = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { register, handleSubmit } = useForm();
+  const onFormSubmit = (data) => {
+    dispatch(signupUser(data));
+    navigate('/');
+  };
```

Comment:
When the form is submitted you're dispatching an action and then using the navigation. However this actions needs to interact with the API so is asynchronous. The navigate function will run before we can signup the user. 

You can see in [this video](https://www.loom.com/share/0549137472f74279b357dddfc407c74e) how we are redirected to the login page for a fraction of a second after the form submit. Please refer to the code on the Login component to fix this behavior.

Reply:
Thank you for the reat explanation😀👍